bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/type/FreeAtHomeChannelGroupTypeProviderImpl.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeathomesystem.internal.type;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.type.ChannelGroupType;
+import org.openhab.core.thing.type.ChannelGroupTypeProvider;
+import org.openhab.core.thing.type.ChannelGroupTypeUID;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ *
+ * @author Andras Uhrin - Initial contribution
+ *
+ */
+@Component(service = { FreeAtHomeChannelGroupTypeProvider.class, ChannelGroupTypeProvider.class })
+public class FreeAtHomeChannelGroupTypeProviderImpl implements FreeAtHomeChannelGroupTypeProvider {
+
+    private final Map<ChannelGroupTypeUID, ChannelGroupType> channelGroupTypeByUID = new HashMap<>();
+
+    @Override
+    public Collection<ChannelGroupType> getChannelGroupTypes(Locale locale) {
+        Collection<ChannelGroupType> result = new ArrayList<>();
+        for (ChannelGroupTypeUID uid : channelGroupTypeByUID.keySet()) {
+            result.add(channelGroupTypeByUID.get(uid));
+        }
+        return result;
+    }
+
+    @Override
+    public @Nullable ChannelGroupType getChannelGroupType(ChannelGroupTypeUID channelGroupTypeUID,
+            @Nullable Locale locale) {
```

Comment:
This parameter is needed because the base class, but not used/needed from the type group generation. The type groups are generated from the text provided from free@home and keywords like r --> read-only.

due to the base class the parameter shall be kept.